cleanup.py

```diff
@@ -0,0 +1,71 @@
+"""
+cleanup docker container/images and mock root for smoke testing
+"""
+
+import logging
+import subprocess
+import re
+
+
+log = logging.getLogger('avocado.test')
+
+def cleanup_docker_and_mock(mockcfg):
+
+    # Clean-up old test artifacts (docker containers, image, mock root)
+
+    docker_containerlist_cmdline = 'docker ps --filter=ancestor=base-runtime-smoke -a -q'
+    try:
+        containerlist = subprocess.check_output(docker_containerlist_cmdline,
+            stderr = subprocess.STDOUT, shell = True)
+    except subprocess.CalledProcessError as e:
+        log.error("command '%s' returned exit status %d; output:\n%s" %
+            (e.cmd, e.returncode, e.output))
+    else:
+        log.info("docker container list with '%s' succeeded with output:\n%s" %
+            (docker_containerlist_cmdline, containerlist))
+
+    if containerlist:
+        containers = re.sub('[\r\n]+', ' ', containerlist)
+        log.info("docker containers using image base-runtime-smoke need to be removed: %s\n" %
+            containers);
+        docker_teardown_cmdline = 'docker rm %s' % containers
+        try:
+            docker_teardown_output = subprocess.check_output(docker_teardown_cmdline,
+                stderr = subprocess.STDOUT, shell = True)
+        except subprocess.CalledProcessError as e:
+            log.error("command '%s' returned exit status %d; output:\n%s" %
+                (e.cmd, e.returncode, e.output))
```

Comment:
If we fail here, we should probably raise an exception that we can catch in the tests so we don't attempt to continue (since the environment isn't clean).